src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -278,6 +284,44 @@ var ReactCompositeComponentMixin = {
     this._pendingReplaceState = false;
     this._pendingForceUpdate = false;
 
+    var markup;
+    if (inst.handleError) {
+      var checkpoint = transaction.checkpoint();
+      try {
+        markup = this.performInitialMount(renderedElement, nativeParent, nativeContainerInfo, transaction, context);
```

Comment:
mostly out of curiosity.

according to: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#2-unsupported-syntax `try`/`catch`/`finally` make their containing function unoptimizable (in V8 at least).

Is there a reason you don't use the suggested workaround of isolating the `try`/`catch`/`finally` statements into minimal functions?